## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 440724`
Commit: `672e1f3124728cf592d564bb1dd8b564d115c02f`

---
### `x86_64-linux`
<details>
  <summary>:x: 14 packages failed to build:</summary>
  <ul>
    <li>aws-adfs (python313Packages.aws-adfs)</li>
    <li>aws-adfs.dist (python313Packages.aws-adfs.dist)</li>
    <li>pyinfra (python313Packages.pyinfra)</li>
    <li>pyinfra.dist (python313Packages.pyinfra.dist)</li>
    <li>python312Packages.aws-adfs</li>
    <li>python312Packages.aws-adfs.dist</li>
    <li>python312Packages.pyinfra</li>
    <li>python312Packages.pyinfra.dist</li>
    <li>python312Packages.trino-python-client</li>
    <li>python312Packages.trino-python-client.dist</li>
    <li>python313Packages.trino-python-client</li>
    <li>python313Packages.trino-python-client.dist</li>
    <li>wapiti</li>
    <li>wapiti.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 178 packages built:</summary>
  <ul>
    <li>atomic-operator</li>
    <li>atomic-operator.dist</li>
    <li>barbicanclient (python313Packages.python-barbicanclient)</li>
    <li>barbicanclient.dist (python313Packages.python-barbicanclient.dist)</li>
    <li>conkeyscan</li>
    <li>conkeyscan.dist</li>
    <li>glanceclient (python313Packages.python-glanceclient)</li>
    <li>glanceclient.dist (python313Packages.python-glanceclient.dist)</li>
    <li>heatclient (python313Packages.python-heatclient)</li>
    <li>heatclient.dist (python313Packages.python-heatclient.dist)</li>
    <li>ironicclient (python313Packages.python-ironicclient)</li>
    <li>ironicclient.dist (python313Packages.python-ironicclient.dist)</li>
    <li>knowsmore</li>
    <li>knowsmore.dist</li>
    <li>magnumclient (python313Packages.python-magnumclient)</li>
    <li>magnumclient.dist (python313Packages.python-magnumclient.dist)</li>
    <li>manilaclient (python313Packages.python-manilaclient)</li>
    <li>manilaclient.dist (python313Packages.python-manilaclient.dist)</li>
    <li>mistralclient (python313Packages.python-mistralclient)</li>
    <li>mistralclient.dist (python313Packages.python-mistralclient.dist)</li>
    <li>nagstamon</li>
    <li>nagstamon.dist</li>
    <li>netexec</li>
    <li>netexec.dist</li>
    <li>neutronclient (python313Packages.python-neutronclient)</li>
    <li>neutronclient.dist (python313Packages.python-neutronclient.dist)</li>
    <li>openstackclient (python313Packages.python-openstackclient)</li>
    <li>openstackclient-full</li>
    <li>openstackclient-full.dist</li>
    <li>openstackclient.dist (python313Packages.python-openstackclient.dist)</li>
    <li>python312Packages.atlassian-python-api</li>
    <li>python312Packages.atlassian-python-api.dist</li>
    <li>python312Packages.dirsearch</li>
    <li>python312Packages.dirsearch.dist</li>
    <li>python312Packages.exchangelib</li>
    <li>python312Packages.exchangelib.dist</li>
    <li>python312Packages.httpie-ntlm</li>
    <li>python312Packages.httpie-ntlm.dist</li>
    <li>python312Packages.httpx-ntlm</li>
    <li>python312Packages.httpx-ntlm.dist</li>
    <li>python312Packages.keystoneauth1</li>
    <li>python312Packages.keystoneauth1.dist</li>
    <li>python312Packages.openstacksdk</li>
    <li>python312Packages.openstacksdk.dist</li>
    <li>python312Packages.openstacksdk.man</li>
    <li>python312Packages.os-client-config</li>
    <li>python312Packages.os-client-config.dist</li>
    <li>python312Packages.osc-lib</li>
    <li>python312Packages.osc-lib.dist</li>
    <li>python312Packages.osc-placement</li>
    <li>python312Packages.osc-placement.dist</li>
    <li>python312Packages.presto-python-client</li>
    <li>python312Packages.presto-python-client.dist</li>
    <li>python312Packages.pypsrp</li>
    <li>python312Packages.pypsrp.dist</li>
    <li>python312Packages.pyspnego</li>
    <li>python312Packages.pyspnego.dist</li>
    <li>python312Packages.pytest-testinfra</li>
    <li>python312Packages.pytest-testinfra.dist</li>
    <li>python312Packages.python-aodhclient</li>
    <li>python312Packages.python-aodhclient.dist</li>
    <li>python312Packages.python-barbicanclient</li>
    <li>python312Packages.python-barbicanclient.dist</li>
    <li>python312Packages.python-cinderclient</li>
    <li>python312Packages.python-cinderclient.dist</li>
    <li>python312Packages.python-designateclient</li>
    <li>python312Packages.python-designateclient.dist</li>
    <li>python312Packages.python-glanceclient</li>
    <li>python312Packages.python-glanceclient.dist</li>
    <li>python312Packages.python-heatclient</li>
    <li>python312Packages.python-heatclient.dist</li>
    <li>python312Packages.python-ironicclient</li>
    <li>python312Packages.python-ironicclient.dist</li>
    <li>python312Packages.python-keystoneclient</li>
    <li>python312Packages.python-keystoneclient.dist</li>
    <li>python312Packages.python-magnumclient</li>
    <li>python312Packages.python-magnumclient.dist</li>
    <li>python312Packages.python-manilaclient</li>
    <li>python312Packages.python-manilaclient.dist</li>
    <li>python312Packages.python-mistralclient</li>
    <li>python312Packages.python-mistralclient.dist</li>
    <li>python312Packages.python-neutronclient</li>
    <li>python312Packages.python-neutronclient.dist</li>
    <li>python312Packages.python-novaclient</li>
    <li>python312Packages.python-novaclient.dist</li>
    <li>python312Packages.python-octaviaclient</li>
    <li>python312Packages.python-octaviaclient.dist</li>
    <li>python312Packages.python-openstackclient</li>
    <li>python312Packages.python-openstackclient.dist</li>
    <li>python312Packages.python-swiftclient</li>
    <li>python312Packages.python-swiftclient.dist</li>
    <li>python312Packages.python-tds</li>
    <li>python312Packages.python-tds.dist</li>
    <li>python312Packages.python-troveclient</li>
    <li>python312Packages.python-troveclient.dist</li>
    <li>python312Packages.python-watcherclient</li>
    <li>python312Packages.python-watcherclient.dist</li>
    <li>python312Packages.python-zaqarclient</li>
    <li>python312Packages.python-zaqarclient.dist</li>
    <li>python312Packages.python-zunclient</li>
    <li>python312Packages.python-zunclient.dist</li>
    <li>python312Packages.pywinrm</li>
    <li>python312Packages.pywinrm.dist</li>
    <li>python312Packages.requests-credssp</li>
    <li>python312Packages.requests-credssp.dist</li>
    <li>python312Packages.requests-kerberos</li>
    <li>python312Packages.requests-kerberos.dist</li>
    <li>python312Packages.requests-ntlm</li>
    <li>python312Packages.requests-ntlm.dist</li>
    <li>python312Packages.smbprotocol</li>
    <li>python312Packages.smbprotocol.dist</li>
    <li>python313Packages.atlassian-python-api</li>
    <li>python313Packages.atlassian-python-api.dist</li>
    <li>python313Packages.dirsearch</li>
    <li>python313Packages.dirsearch.dist</li>
    <li>python313Packages.exchangelib</li>
    <li>python313Packages.exchangelib.dist</li>
    <li>python313Packages.httpie-ntlm</li>
    <li>python313Packages.httpie-ntlm.dist</li>
    <li>python313Packages.httpx-ntlm</li>
    <li>python313Packages.httpx-ntlm.dist</li>
    <li>python313Packages.keystoneauth1</li>
    <li>python313Packages.keystoneauth1.dist</li>
    <li>python313Packages.openstacksdk</li>
    <li>python313Packages.openstacksdk.dist</li>
    <li>python313Packages.openstacksdk.man</li>
    <li>python313Packages.os-client-config</li>
    <li>python313Packages.os-client-config.dist</li>
    <li>python313Packages.osc-lib</li>
    <li>python313Packages.osc-lib.dist</li>
    <li>python313Packages.osc-placement</li>
    <li>python313Packages.osc-placement.dist</li>
    <li>python313Packages.presto-python-client</li>
    <li>python313Packages.presto-python-client.dist</li>
    <li>python313Packages.pypsrp</li>
    <li>python313Packages.pypsrp.dist</li>
    <li>python313Packages.pyspnego</li>
    <li>python313Packages.pyspnego.dist</li>
    <li>python313Packages.pytest-testinfra</li>
    <li>python313Packages.pytest-testinfra.dist</li>
    <li>python313Packages.python-aodhclient</li>
    <li>python313Packages.python-aodhclient.dist</li>
    <li>python313Packages.python-cinderclient</li>
    <li>python313Packages.python-cinderclient.dist</li>
    <li>python313Packages.python-designateclient</li>
    <li>python313Packages.python-designateclient.dist</li>
    <li>python313Packages.python-keystoneclient</li>
    <li>python313Packages.python-keystoneclient.dist</li>
    <li>python313Packages.python-novaclient</li>
    <li>python313Packages.python-novaclient.dist</li>
    <li>python313Packages.python-octaviaclient</li>
    <li>python313Packages.python-octaviaclient.dist</li>
    <li>swiftclient (python313Packages.python-swiftclient)</li>
    <li>swiftclient.dist (python313Packages.python-swiftclient.dist)</li>
    <li>python313Packages.python-tds</li>
    <li>python313Packages.python-tds.dist</li>
    <li>troveclient (python313Packages.python-troveclient)</li>
    <li>troveclient.dist (python313Packages.python-troveclient.dist)</li>
    <li>watcherclient (python313Packages.python-watcherclient)</li>
    <li>watcherclient.dist (python313Packages.python-watcherclient.dist)</li>
    <li>python313Packages.python-zaqarclient</li>
    <li>python313Packages.python-zaqarclient.dist</li>
    <li>zunclient (python313Packages.python-zunclient)</li>
    <li>zunclient.dist (python313Packages.python-zunclient.dist)</li>
    <li>python313Packages.pywinrm</li>
    <li>python313Packages.pywinrm.dist</li>
    <li>python313Packages.requests-credssp</li>
    <li>python313Packages.requests-credssp.dist</li>
    <li>python313Packages.requests-kerberos</li>
    <li>python313Packages.requests-kerberos.dist</li>
    <li>python313Packages.requests-ntlm</li>
    <li>python313Packages.requests-ntlm.dist</li>
    <li>python313Packages.smbprotocol</li>
    <li>python313Packages.smbprotocol.dist</li>
    <li>sccmhunter</li>
    <li>sccmhunter.dist</li>
    <li>trevorspray</li>
    <li>trevorspray.dist</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `x86_64-linux`</summary>
<details>
<summary>aws-adfs</summary>
<pre>Running phase: buildPhase
@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;buildPhase&quot; }
Executing pypaBuildPhase
Creating a wheel...
pypa build flags: --no-isolation --outdir dist/ --wheel
* Getting build dependencies for wheel...
* Building wheel...
Successfully built aws_adfs-2.12.0-py3-none-any.whl
Finished creating a wheel...
/build/source/dist /build/source
Unpacking to: unpacked/aws_adfs-2.12.0...OK
Repacking wheel as ./aws_adfs-2.12.0-py3-none-any.whl...OK
/build/source
Finished executing pypaBuildPhase
Running phase: pythonRuntimeDepsCheckHook
@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;pythonRuntimeDepsCheckHook&quot; }
Executing pythonRuntimeDepsCheck
Checking runtime dependencies for aws_adfs-2.12.0-py3-none-any.whl
  - fido2&lt;2.0,&gt;=1.0 not satisfied by version 2.0.0
  - lxml&lt;6.0,&gt;=4.9 not satisfied by version 6.0.0</pre>
</details>
<details>
<summary>pyinfra</summary>
<pre>adding &#x27;tests/test_connectors/test_local.py&#x27;
adding &#x27;tests/test_connectors/test_ssh.py&#x27;
adding &#x27;tests/test_connectors/test_sshuserclient.py&#x27;
adding &#x27;tests/test_connectors/test_terraform.py&#x27;
adding &#x27;tests/test_connectors/test_util.py&#x27;
adding &#x27;tests/test_connectors/test_vagrant.py&#x27;
adding &#x27;pyinfra-3.4.1.dist-info/METADATA&#x27;
adding &#x27;pyinfra-3.4.1.dist-info/WHEEL&#x27;
adding &#x27;pyinfra-3.4.1.dist-info/entry_points.txt&#x27;
adding &#x27;pyinfra-3.4.1.dist-info/top_level.txt&#x27;
adding &#x27;pyinfra-3.4.1.dist-info/RECORD&#x27;
removing build/bdist.linux-x86_64/wheel
Successfully built pyinfra-3.4.1-py2.py3-none-any.whl
Finished creating a wheel...
Finished executing pypaBuildPhase
Running phase: pythonRuntimeDepsCheckHook
@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;pythonRuntimeDepsCheckHook&quot; }
Executing pythonRuntimeDepsCheck
Checking runtime dependencies for pyinfra-3.4.1-py2.py3-none-any.whl
  - paramiko&lt;4,&gt;=2.7 not satisfied by version 4.0.0</pre>
</details>
<details>
<summary>python312Packages.trino-python-client</summary>
<pre>@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;pythonImportsCheckPhase&quot; }
Executing pythonImportsCheckPhase
Check whether the following modules can be imported: trino
Traceback (most recent call last):
  File &quot;&lt;string&gt;&quot;, line 1, in &lt;module&gt;
  File &quot;&lt;string&gt;&quot;, line 1, in &lt;lambda&gt;
  File &quot;/nix/store/18l19gnl8dq75v6knghbcgkrikmpc0yv-python3-3.12.11/lib/python3.12/importlib/__init__.py&quot;, line 90, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File &quot;&lt;frozen importlib._bootstrap&gt;&quot;, line 1387, in _gcd_import
  File &quot;&lt;frozen importlib._bootstrap&gt;&quot;, line 1360, in _find_and_load
  File &quot;&lt;frozen importlib._bootstrap&gt;&quot;, line 1331, in _find_and_load_unlocked
  File &quot;&lt;frozen importlib._bootstrap&gt;&quot;, line 935, in _load_unlocked
  File &quot;&lt;frozen importlib._bootstrap_external&gt;&quot;, line 999, in exec_module
  File &quot;&lt;frozen importlib._bootstrap&gt;&quot;, line 488, in _call_with_frames_removed
  File &quot;/nix/store/l912a8jczv7c7ig3q2asbyz524ya3bgj-python3.12-trino-python-client-0.334.0/lib/python3.12/site-packages/trino/__init__.py&quot;, line 13, in &lt;module&gt;
    from . import client
  File &quot;/nix/store/l912a8jczv7c7ig3q2asbyz524ya3bgj-python3.12-trino-python-client-0.334.0/lib/python3.12/site-packages/trino/client.py&quot;, line 68, in &lt;module&gt;
    import lz4.block
ModuleNotFoundError: No module named &#x27;lz4&#x27;</pre>
</details>
<details>
<summary>python313Packages.trino-python-client</summary>
<pre>  File &quot;&lt;string&gt;&quot;, line 1, in &lt;module&gt;
    import sys; import importlib; list(map(lambda mod: importlib.import_module(mod), sys.argv[1:]))
                                  ~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File &quot;&lt;string&gt;&quot;, line 1, in &lt;lambda&gt;
    import sys; import importlib; list(map(lambda mod: importlib.import_module(mod), sys.argv[1:]))
                                                       ~~~~~~~~~~~~~~~~~~~~~~~^^^^^
  File &quot;/nix/store/iyff8129iampdw13nlfqalzhxy8y1hi9-python3-3.13.6/lib/python3.13/importlib/__init__.py&quot;, line 88, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           ~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File &quot;&lt;frozen importlib._bootstrap&gt;&quot;, line 1387, in _gcd_import
  File &quot;&lt;frozen importlib._bootstrap&gt;&quot;, line 1360, in _find_and_load
  File &quot;&lt;frozen importlib._bootstrap&gt;&quot;, line 1331, in _find_and_load_unlocked
  File &quot;&lt;frozen importlib._bootstrap&gt;&quot;, line 935, in _load_unlocked
  File &quot;&lt;frozen importlib._bootstrap_external&gt;&quot;, line 1026, in exec_module
  File &quot;&lt;frozen importlib._bootstrap&gt;&quot;, line 488, in _call_with_frames_removed
  File &quot;/nix/store/zaykb8jpm8jr40s22i21vqibwkvlp1b4-python3.13-trino-python-client-0.334.0/lib/python3.13/site-packages/trino/__init__.py&quot;, line 13, in &lt;module&gt;
    from . import client
  File &quot;/nix/store/zaykb8jpm8jr40s22i21vqibwkvlp1b4-python3.13-trino-python-client-0.334.0/lib/python3.13/site-packages/trino/client.py&quot;, line 68, in &lt;module&gt;
    import lz4.block
ModuleNotFoundError: No module named &#x27;lz4&#x27;</pre>
</details>
<details>
<summary>wapiti</summary>
<pre>        &lt;textarea&gt;
        &lt;/noscript&gt;
        &lt;/body&gt;
        &lt;/html&gt;&quot;&quot;&quot;
    
&gt;       assert get_context_list(html, &quot;injection&quot;) == [
            {&quot;non_exec_parent&quot;: &quot;noscript&quot;, &quot;parent&quot;: &quot;textarea&quot;, &quot;type&quot;: &quot;comment&quot;}
        ]
E       AssertionError: assert [{&#x27;non_exec_p...ype&#x27;: &#x27;text&#x27;}] == [{&#x27;non_exec_p...&#x27;: &#x27;comment&#x27;}]
E         
E         At index 0 diff: {&#x27;type&#x27;: &#x27;text&#x27;, &#x27;parent&#x27;: &#x27;textarea&#x27;, &#x27;non_exec_parent&#x27;: &#x27;noscript&#x27;} != {&#x27;non_exec_parent&#x27;: &#x27;noscript&#x27;, &#x27;parent&#x27;...
E         
E         ...Full output truncated (2 lines hidden), use &#x27;-vv&#x27; to show

tests/parsers/test_xss_utils.py:82: AssertionError
=========================== short test summary info ============================
FAILED tests/parsers/test_xss_utils.py::test_title_context - AssertionError: assert [{&#x27;non_exec_p...ype&#x27;: &#x27;text&#x27;}] == [{&#x27;non_exec_p...yp...
FAILED tests/parsers/test_xss_utils.py::test_noscript_context - assert [{&#x27;non_exec_p...ype&#x27;: &#x27;text&#x27;}] == [{&#x27;events&#x27;: s...r&#x27;: &#x27;&quot;&#x27;, ...}]
FAILED tests/parsers/test_xss_utils.py::test_comment_in_noscript_context - AssertionError: assert [{&#x27;non_exec_p...ype&#x27;: &#x27;text&#x27;}] == [{&#x27;non_exec_p...&#x27;:...
================ 3 failed, 239 passed, 133 deselected in 13.63s ================</pre>
</details>
</details>

--- a/pkgs/development/python-modules/aws-adfs/default.nix
+++ b/pkgs/development/python-modules/aws-adfs/default.nix
@@ -37,6 +37,8 @@ buildPythonPackage rec {
 
   pythonRelaxDeps = [
     "configparser"
+    "fido2"
+    "lxml"
     "requests-kerberos"
   ];
 

--- a/pkgs/development/python-modules/trino-python-client/default.nix
+++ b/pkgs/development/python-modules/trino-python-client/default.nix
@@ -1,26 +1,28 @@
 {
   lib,
   buildPythonPackage,
+  boto3,
   fetchFromGitHub,
-  # build
-  setuptools,
-  # required
-  pytz,
-  requests,
-  tzlocal,
-  # optional
-  requests-kerberos,
-  sqlalchemy,
-  keyring,
-  # tests
-  pytestCheckHook,
   httpretty,
+  keyring,
+  lz4,
+  pytestCheckHook,
+  python-dateutil,
+  pytz,
+  requests-gssapi,
+  requests-kerberos,
+  requests,
+  setuptools,
+  sqlalchemy,
+  testcontainers,
+  tzlocal,
+  zstandard,
 }:
 
 buildPythonPackage rec {
   pname = "trino-python-client";
   version = "0.334.0";
-  format = "setuptools";
+  pyproject = true;
 
   src = fetchFromGitHub {
     repo = "trino-python-client";
@@ -29,37 +31,50 @@ buildPythonPackage rec {
     hash = "sha256-cSwMmzIUFYX8VgSwobth8EsARUff3hhfBf+IrhuFSYM=";
   };
 
-  nativeBuildInputs = [ setuptools ];
+  build-system = [ setuptools ];
 
-  propagatedBuildInputs = [
+  dependencies = [
+    lz4
+    python-dateutil
     pytz
     requests
     tzlocal
+    zstandard
   ];
 
   optional-dependencies = lib.fix (self: {
     kerberos = [ requests-kerberos ];
+    gsaapi = [ requests-gssapi ];
     sqlalchemy = [ sqlalchemy ];
     external-authentication-token-cache = [ keyring ];
     all = self.kerberos ++ self.sqlalchemy;
   });
 
   nativeCheckInputs = [
+    boto3
     httpretty
     pytestCheckHook
+    testcontainers
   ]
-  ++ optional-dependencies.all;
+  ++ lib.flatten (builtins.attrValues optional-dependencies);
 
   pythonImportsCheck = [ "trino" ];
 
   disabledTestPaths = [
-    # these all require a running trino instance
+    # Tests require a running trino instance
     "tests/integration/test_types_integration.py"
     "tests/integration/test_dbapi_integration.py"
     "tests/integration/test_sqlalchemy_integration.py"
   ];
 
   disabledTestMarks = [ "auth" ];
+
+  disabledTests = [
+    # Tests require a running trino instance
+    "test_oauth2"
+    "test_token_retrieved_once_when_authentication_instance_is_shared"
+    "test_multithreaded_oauth2_authentication_flow"
+  ];
 
   meta = with lib; {
     changelog = "https://github.com/trinodb/trino-python-client/blob/${src.tag}/CHANGES.md";

